Mathlib/Tactic/Set.lean

```diff
@@ -53,11 +53,10 @@ elab_rules : tactic
       pure (fvar, [goal])
     Term.addTermInfo' (isBinder := true) a (mkFVar fvar)
     if rw.isNone then
-      evalTactic (← `(tactic| try rewrite [(id rfl : $val = $a)] at *))
-    let tt ← Term.exprToSyntax ty
+      evalTactic (← `(tactic| try rewrite [show $(← Term.exprToSyntax vale) = $a from rfl] at *))
     match h, rev with
     | some h, some none =>
-      evalTactic (← `(tactic| have%$tk $h : $a = ($val : $tt) := rfl))
+      evalTactic (← `(tactic| have%$tk $h : $a = ($val : $(← Term.exprToSyntax ty)) := rfl))
```

Comment:
Here and two lines later, should we do this?
```suggestion
      evalTactic (← `(tactic| have%$tk $h : $a = ($(← Term.exprToSyntax vale) : $(← Term.exprToSyntax ty)) := rfl))
```
Another option is that right after line 54 you do
```lean
let val ← Term.exprToSyntax vale
let ty ← Term.exprToSyntax ty
```
so that you can use `$val` and `$ty` in these quotations.

Reply:
Oh yes that's probably better thanks